core/src/client/async_client/mod.rs

```diff
@@ -114,8 +115,13 @@ impl ClientBuilder {
 	///
 	/// ## Panics
 	///
-	/// Panics if being called outside of `tokio` runtime context.
-	pub fn build<S: TransportSenderT, R: TransportReceiverT>(self, sender: S, receiver: R) -> Client {
+	/// Panics if called outside of `tokio` runtime context.
+	#[cfg(feature = "async-client")]
+	pub fn build_with_tokio<S, R>(self, sender: S, receiver: R) -> Client
```

Comment:
ideally I would get rid of these but that would require some abstraction over the executor which I haven't found